lib/Doctrine/DBAL/Schema/Column.php

```diff
@@ -94,106 +90,59 @@ public function setOptions(array $options)
         return $this;
     }
 
-    /**
-     * @return Column
-     */
-    public function setType(Type $type)
+    public function setType(Type $type) : self
     {
         $this->_type = $type;
 
         return $this;
     }
 
-    /**
-     * @param int|null $length
-     *
-     * @return Column
-     */
-    public function setLength($length)
+    public function setLength(?int $length) : self
     {
-        if ($length !== null) {
-            $this->_length = (int) $length;
-        } else {
-            $this->_length = null;
-        }
+        $this->_length = $length;
 
         return $this;
     }
 
-    /**
-     * @param int $precision
-     *
-     * @return Column
-     */
-    public function setPrecision($precision)
+    public function setPrecision(?int $precision) : self
     {
-        if (! is_numeric($precision)) {
-            $precision = 10; // defaults to 10 when no valid precision is given.
-        }
-
-        $this->_precision = (int) $precision;
+        $this->_precision = $precision;
```

Comment:
Should we check for `$precision >= 0`?

Reply:
I don't think so:

1. It's out of scope.
2. The class is just a DTO, not a domain object. DBAL doesn't do any handling of the precision AFAIK, and the DB will validate it itself.
3. In the domain, the value is not only bounded from at bottom (is it?), it also may be bounded at the top which we know nothing about at the abstraction layer.
4. The max value may be dependent on other values (e.g. on Oracle, the sum of precision and scale can't be greater than 38).